lib/datadog/core/normalizer.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+module Datadog
+  module Core
+    module Normalizer
+      module_function
+
+      INVALID_TAG_CHARACTERS = %r{[^a-z0-9_\-:./]}.freeze
+
+      # Based on https://docs.datadoghq.com/getting_started/tagging/#defining-tags
+      # Currently a reimplementation of the logic in the
+      # Datadog::Tracing::Metadata::Ext::HTTP::Headers.to_tag method with some additional items
+      # TODO: Swap out the logic in the Datadog Tracing Metadata headers logic
+      def self.normalize(original_value)
+        return "" if original_value.nil? || original_value.to_s.strip.empty?
+
+        # Removes whitespaces
+        normalized_value = original_value.to_s.strip
+        # Lower case characters
+        normalized_value.downcase!
+        # Invalid characters are replaced with an underscore
+        normalized_value.gsub!(INVALID_TAG_CHARACTERS, '_')
+        # Merge consecutive underscores with a single underscore
+        normalized_value.squeeze!('_')
```

Comment:
```suggestion
```

Actually I don't believe we're supposed to do the squeeze (reduce multiple `_` characters to a single character). Or at least the spec doesn't say to do so. Here the Python tracer leaves repeated underscores:

https://github.com/DataDog/dd-trace-py/pull/15146/files#diff-734f80f7c77b609471c1ca40c131a2604c2f50647ad9cf264863e2016f07b209R23

We should remove this to be consistent.